beater/processors.go

```diff
@@ -57,3 +61,37 @@ func rateLimitBatchProcessor(ctx context.Context, batch *model.Batch) error {
 	}
 	return nil
 }
+
+// ecsVersionBatchProcessor is a model.BatchProcessor that sets the ECSVersion
+// field of each event to the ECS library version.
+func ecsVersionBatchProcessor(ctx context.Context, b *model.Batch) error {
+	for i := range *b {
+		event := &(*b)[i]
+		event.ECSVersion = ecs.Version
+	}
+	return nil
+}
+
+// newBeatInfoBatchProcessor returns a model.BatchProcessor that sets observer
+// fields from info.
+func newBeatInfoBatchProcessor(info beat.Info) model.ProcessBatchFunc {
```

Comment:
How about calling this `observerBatchProcessor`?

Reply:
Yes, that is a better name. Done.